src/app/public/modules/infinite-scroll/infinite-scroll.component.ts

```diff
@@ -45,24 +51,28 @@ export class SkyInfiniteScrollComponent implements OnInit, OnDestroy {
   public ngOnInit(): void {
     // The user has scrolled to the infinite scroll element.
     this.domAdapter.scrollTo(this.elementRef)
```

Comment:
From an efficiency standpoint would it be better to set these up whenever the infinite scroll is enabled each time? With this these observables will fire even when not enabled and they just won't do anything if enabled. If we moved this setup to fire when `enabled` is set to true then we wouldn't have them firing when disabled.

I also have a similar concern about the fact that the `domAdapter` has events that fire on all scrolls during the lifecycle of the adapter and not just when something is listening to it but that is a separate can of worms :sweat_smile:

Reply:
Yeah, that would be a much better implementation. Let me dig into that can real quick and see what I find 😉

Reply:
Not so bad. Should be more efficient now. Great suggestion, Trevor! 😄

Reply:
Looks great! Thanks!